hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkTaskContextSupplier.java

```diff
@@ -90,4 +121,18 @@ public Option<String> getProperty(EngineProperty prop) {
     throw new HoodieException("Unknown engine property :" + prop);
   }
 
+  @Override
+  public Option<HoodieReaderContext> getReaderContext(HoodieTableMetaClient metaClient, boolean useReaderContext) {
```

Comment:
this is a engine level class.. and this method feels  out of place.

Reply:
Right now the `HoodieReaderContext` is designed to be engine-specific.  Ideally it should be in `HoodieEngineContext`.  We've created `EngineBroadcastManager` to retrieve the reader context.  HUDI-8637 as the follow-up.

Reply:
@yihua , I can take it to the HoodieEngineContext while I am fixing other bugs. 
@danny0405 , do you have any other objections?